nucypher/blockchain/economics.py

```diff
@@ -14,60 +14,166 @@
+    """
+
+    # Token Denomination
+    __token_decimals = 18
+    nunits_per_token = 10 ** __token_decimals          # Smallest unit designation
+
+    # Period Definition
+    hours_per_period = 24                            # Hours in single period
+    seconds_per_period = hours_per_period * 60 * 60  # Seconds in single period
+
+    # Time Constraints
+    minimum_locked_periods = 30                      # 720 Hours minimum
+
+    # Value Constraints
+    minimum_allowed_locked = 15000 * nunits_per_token
+    maximum_allowed_locked = int(4e6) * nunits_per_token
+
+    # Supply
+    __default_initial_supply = Decimal(int(1e9) * nunits_per_token)
+    __default_initial_inflation = Decimal(1.0)
+    __default_token_halving = Decimal(2.0)     # years
+    __default_reward_saturation = Decimal(1.0)  # years
+    __default_small_stake_multiplier = Decimal(0.5)
+
+    def __init__(self,
+                 initial_supply: Decimal = __default_initial_supply,
+                 initial_inflation: Decimal = __default_initial_inflation,
+                 halving_delay: Decimal = __default_token_halving,
+                 reward_saturation: Decimal = __default_reward_saturation,
+                 small_stake_multiplier: Decimal = __default_small_stake_multiplier
+                 ):
         """
-        Calculate parameters to use in token issuer from high-level
-        human-understandable parameters
-
-        :param initial_supply: coins at t=0,
-        :param initial_inflation: inflation in day 1 expressed in units of year**-1
-            e.g. 1/365 of all coins in day 1 is 1 year**-1
-        :param T_half: time for inflation halving in years
-        :param T_sat: "saturation" time - if staking is longer than T_sat, the
-            reward doesn't get higher
-        :param small_staker_multiplier: what fraction of max reward rate do we
-            pay to those who are about to unlock coins
+        :param initial_supply: Tokens at t=0
+        :param initial_inflation: Inflation on day 1 expressed in units of year**-1
+        :param halving_delay: Time for inflation halving in years
+        :param reward_saturation: "saturation" time - if staking is longer than T_sat, the reward doesn't get any higher
+        :param small_stake_multiplier: Fraction of maximum reward rate paid to those who are about to unlock tokens
         """
 
-        # Saving input in case we'll need it
-        self.input_params = dict(
-                initial_supply=initial_supply,
-                initial_inflation=initial_inflation,
-                T_half=T_half,
-                T_sat=T_sat,
-                small_staker_multiplier=small_staker_multiplier)
+        self.initial_supply = initial_supply
+        self.initial_inflation = initial_inflation
+        self.token_halving = halving_delay
+        self.token_saturation = reward_saturation
+        self.small_stake_multiplier = small_stake_multiplier
+        self.log_scale = Decimal(log(2))
```

Comment:
I don't see how `self.log_scale` is more readable than `log(2)`. I'd much rather use the latter. Otherwise, it's like using `self.two` instead of `2`.
Of course, it could be not "halving" delay, but delay for decreasing by any factor... But we adopted halving in variable naming. So why not leave it as `log(2)`?

Reply:
Ahh, I see, because it's decimal. I'd call it `log2` then

Reply:
:heavy_check_mark: